File: C#/DataStructures/Test/sorts/BubbleSortTest.cs
using DataStructures.sorts;
using DataStructures.util;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Test.util;

namespace Test.sorts
{
    [TestClass]
    public class BubbleSortTest
    {
        [TestMethod]
        public void BubbleSort_SortsCorrectly()
        {
            var input = TestUtil.RandomArray(TestUtil.ARR_SIZE);

            var result = BubbleSort.Sort(input);

            Assert.IsTrue(SortUtil.IsSorted(result));
        }
    }
}

File: Scala/data-structures/src/test/scala/sorts/BubbleSortTest.scala
package sorts

import com.ckeyser.sorts.BubbleSort
import org.scalatest.FunSpec
import com.ckeyser.util.{ArrayUtil, SortUtil}

/**
  * Created by Chris Keyser on 9/28/2016.
  */
class BubbleSortTest extends FunSpec {
    it("should correctly sort an array") {
        val input = ArrayUtil.randomArray(ArrayUtil.DefaultSize)

        val result = BubbleSort.sort(input)

        assert(SortUtil.isSorted(result))
    }
}
